packages/cloud_firestore/android/src/main/java/io/flutter/plugins/firebase/cloudfirestore/CloudFirestorePlugin.java

```diff
@@ -245,6 +245,30 @@ public void onMethodCall(MethodCall call, final Result result) {
           addDefaultListeners("updateData", task, result);
           break;
         }
+      case "DocumentReference#get":
+        {
+          Map<String, Object> arguments = call.arguments();
+          DocumentReference documentReference = getDocumentReference(arguments);
+          Task<DocumentSnapshot> task = documentReference.get();
+          task.addOnSuccessListener(
+                  new OnSuccessListener<DocumentSnapshot>() {
+                    @Override
+                    public void onSuccess(DocumentSnapshot documentSnapshot) {
+                      Map<String, Object> snapshotMap = new HashMap<>();
```

Comment:
Chiming in here as I was trying this patch, but I don't think this works as expected for a document that doesn't exist. I get:

```
java.lang.IllegalStateException: This document doesn't exist. Use DocumentSnapshot.exists() to check whether the document exists before accessing its fields.
```

I think this needs to look something like:

```java
                    public void onSuccess(DocumentSnapshot documentSnapshot) {
                      Map<String, Object> snapshotMap = new HashMap<>();
                      snapshotMap.put("path", documentSnapshot.getReference().getPath());
                      if (documentSnapshot.exists()) {
                        snapshotMap.put("data", documentSnapshot.getData());
                      } else {
                        snapshotMap.put("data", null);
                      }
                      result.success(snapshotMap);
                    }
```

Reply:
@oliverchang good catch! I'll fix.